pulp_puppet_plugins/pulp_puppet/plugins/migrations/0005_puppet_module_name_change.py

```diff
@@ -0,0 +1,59 @@
+from gettext import gettext as _
+import logging
+
+from mongoengine import NotUniqueError, Q
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.db import model
+
+from pulp_puppet.plugins.db.models import Module
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    For each puppet module check and if needed update module name format.
+
+    There was a discrepancy in the way puppet module's name was stored in pulp, depending
+    if it was synced from filesystem or uploaded. This migration finds puppet module units
+    that have wrong format name and replaces it with a correct format name.
+    """
+
+    modules = Module.objects(Q(name__contains='/') | Q(name__contains='-'))
+    repos_to_rebuild = set()
+    for puppet_unit in modules:
+        try:
+            author, name = puppet_unit['name'].split("-", 1)
+        except ValueError:
+            # This is the forge format, but Puppet still allows it
+            author, name = puppet_unit['name'].split("/", 1)
+        try:
+            puppet_unit.name = name
```

Comment:
This could be outside of the `try` block since it's not part of the exception being handled.